docs/proposals/20220712-cluster-api-addon-orchestration.md

```diff
@@ -0,0 +1,448 @@
+
+### Implementation details 
+
+The implementation will include two CRDs and two controllers as follows. As this is a prototype, the specific details regarding the CRD specs are subject to change.
+
+#### HelmChartProxy CRD
+
+The first CRD, HelmChartProxy, serves as the user interface for defining and configuring Helm charts as well as selecting which workload clusters it will be installed on. 
+
+```go
+// HelmChartProxySpec defines the desired state of HelmChartProxy.
+type HelmChartProxySpec struct {
+	// ClusterSelector selects Clusters with a label that matches the specified key/value pair. The Helm chart will be
+	// installed on all selected Clusters. If a Cluster is no longer selected, the Helm release will be uninstalled.
+	ClusterSelector ClusterSelectorLabel `json:"clusterSelector"`
+
+	// ChartName is the name of the Helm chart in the repository.
+	ChartName string `json:"chartName,omitempty"`
+
+	// RepoURL is the URL of the Helm chart repository.
+	RepoURL string `json:"repoURL,omitempty"`
+
+	// ReleaseName is the release name of the installed Helm chart. If it is not specified, a name will be generated.
+	// +optional
+	ReleaseName string `json:"releaseName,omitempty"`
+
+	// Version is the version of the Helm chart. If it is not specified, the latest version will be used.
+	// +optional
+	Version string `json:"version,omitempty"`
+
+	// Namespace is the namespace the Helm release will be installed on each selected Cluster. If it is not specified, the default namespace.
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+
+	// Values is an inline YAML representing the values for the Helm chart. This YAML supports Go templating to reference
+	// fields from each selected workload Cluster and programatically create and set values.
+	// +optional
+	Values string `json:"values,omitempty"`
+}
+
+// ClusterSelectorLabel defines a key/value pair used to select Clusters with a label matching the specified key and value.
+type ClusterSelectorLabel struct {
+	// Key is the label key.
+	Key string `json:"key"`
+
+	// Value is the label value.
+	Value string `json:"value"`
+}
+
+// HelmChartProxyStatus defines the observed state of HelmChartProxy.
+type HelmChartProxyStatus struct {
+	// Ready is true when the HelmReleaseProxySpec for each selected Cluster is up to date.
+	// +optional
+	Ready bool `json:"ready"`
+
+	// MatchingClusters is the list of references to Clusters selected by the ClusterSelectorLabel.
+	// +optional
+	MatchingClusters []corev1.ObjectReference `json:"matchingClusters"`
+
+	// Conditions defines current state of the HelmChartProxy.
+	// +optional
+	Conditions clusterv1.Conditions `json:"conditions,omitempty"`
+
+	// FailureReason will be set in the event that there is a an error reconciling the HelmChartProxy.
+	// +optional
+	FailureReason string `json:"failureReason,omitempty"`
```

Comment:
why do you need this over conditions?

Reply:
This was paired with the `ready` bool, I think we can probably just drop them both in that case